### ISSUE TITLE
ENH: spatial color butterfly plots

### ISSUE THREAD

**Issue by jona-sassenhagen** [mne-tools/mne-python#2533]
This is how I plot:
![unknown](https://cloud.githubusercontent.com/assets/4321826/10540836/4bcd9962-73d1-11e5-9b66-6e601899d9fb.png)

- butterfly style
- channels are color-coded by mapping their x/y/z positions into RGB space (see legend in plot)
- combine butterfly and topomaps, EEGLAB style

I think this is a very good way of visualizing spatio-temporal dynamics. I've discussed this with @dengemann before and IIRC he liked it.

Color coding is fairly straight-forward and I'd like to begin with that. The code I used above is simply:
```
                if spatial_colors:
                    chspace = np.asarray([list(chs[ch]['loc'][0:3]) 
                                          for ch in range(len(chs))])
                    chspace -= chspace.min()
                    chspace /= chspace.max()
                    ax._get_lines.color_cycle = iter(chspace)
```

in `_plot_evoked`. Can I implement this? My PR schedule is comparatively empty right now.
(Topomaps and the in-plot legend are a bit more complicated, but this part is fairly simple.)

**Comment by agramfort** [comment]
nice. why not

I am sure @kingjr will like this :)

**Comment by Eric89GXL** [comment]
Sure. I wonder if we could do something more systematic with the colors, though, like have it go blue on the left to red on the right, with value from light at the top to dark at the bottom. That way, given a color in the plot, you immediately have a sense of where it is without having to look at the key.

**Comment by jona-sassenhagen** [comment]
@dengemann said we should maybe find a different color map.

**Comment by kingjr** [comment]
yes I really like the idea. I would do a 2D transform though, as the sensors are on a surface, not a volume. Have you tried using a double gradient : e.g. red/blue for coding left-right and black/white for front-back?

**Comment by jona-sassenhagen** [comment]
@kingjr @Eric89GXL haha

**Comment by Eric89GXL** [comment]
@jona-sassenhagen there you go, red/blue light/dark wins sounds like :)

**Comment by kingjr** [comment]
great minds :]

**Comment by jona-sassenhagen** [comment]
closed for #2638
